### DISCUSSION TITLE
Site-wide annoucement `banner`

### DISCUSSION THREAD

**Discussion by wkillerud** [withastro/starlight#1535] [Feature Requests]
### What version of `starlight` are you using?

0.19.0

### What is your idea?

The [banner frontmatter](https://starlight.astro.build/reference/frontmatter/#banner), but configured in one location to set the banner for all pages.

### Why is this feature necessary?

I'm building a new docsite using Starlight, and thought I would add this banner to set expectations and maybe nudge contributors to provide feedback or changes:

![Banner saying This documentation is in beta, and we're looking for feedback. See how you can contribute.](https://github.com/withastro/starlight/assets/1223410/ccd40e4b-b7d4-4cf5-b7dd-c3c422f44854)

I can add the banner to all pages (since folks can land on any page), but that got me looking for a central location to add it instead.

I found a serviceable workaround using a [component override for MarkdownContent](https://starlight.astro.build/reference/overrides/#markdowncontent), but it doesn't look quite as nice rendered as part of the main contents.

![The same banner, placed below the title and above the main contents](https://github.com/withastro/starlight/assets/1223410/61e0eed0-5177-461e-b227-157fd92d702e)

I imagine a config in `starlight()` `defaultBanner`, or something along those lines. To opt out individual pages, maybe set the banner frontmatter content to an empty string?

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Hi @wkillerud!

Thanks for the feedback.

One quick workaround: you can provide default frontmatter values by extending the `docsSchema()` in `src/content/config.ts`:

```ts
// src/content/config.ts
import { defineCollection, z } from 'astro:content';
import { docsSchema } from '@astrojs/starlight/schema';

export const collections = {
	docs: defineCollection({
		schema: docsSchema({
			extend: z.object({
				banner: z.object({ content: z.string() }).default({
					content:
						'This documentation is in beta, and we’re looking for feedback. See how <a href="#">you can contribute</a> 🌟',
				}),
			}),
		}),
	}),
};
```

This will then apply to all pages that don’t have their own `banner` config.

When building the banner feature we did initially consider building it as a top-level rather than page-level feature but had a couple of reservations:

1. Site-wide banners can easily be annoying grabbing a lot of attention, so if we were to support them, making them dismissible seemed important. But that’s also more complex, so we skipped it at the time 😁 (Not saying this is your case — an “in construction” banner seems helpful to flag everywhere.)

2. Starlight’s header design doesn’t easily support slotting extra stuff up top (often these site-wide banners are dismissible and right at the top of the page before the nav bar). So we decided to slot the banner in this location:

	<img width="1193" alt="Starlight docs with a banner showing in the main content column" src="https://github.com/withastro/starlight/assets/357379/98ae4da3-81b3-4a6f-accd-bef47451b310">

	Given this looks visually more like “part of the page”, the config at that level made sense too.

4. We always try to consider localization when building features and weren’t confident yet on a site-wide config that would support setting banner messages in all languages a site uses, so punted on that — setting at the page level means this is handled “automatically” as a translated page provides a translated banner in frontmatter.

I still think it could be nice to support the dismissible site-wide banner pattern, but wanted to share some of the reasons we didn’t yet and things we’d want to solve if we were to do this.

**Reply by wkillerud** [reply]
> One quick workaround: you can provide default frontmatter values by extending the `docsSchema()` in `src/content/config.ts`

Oh nice, that's perfect – thank you 🙏 

> [...] had a couple of reservations

Totally valid, I figured it would be more complicated than just adding a new setting 😄 Thank you for sharing!

**Comment by trueberryless** [comment]
As mentioned in https://github.com/withastro/starlight/issues/1748#issuecomment-2831180285 I think adding a global "announcement" banner would be a nice addition to Starlight.
<img width="3580" height="474" alt="image" src="https://github.com/user-attachments/assets/1b1be3f9-c430-4353-9bf5-7c4dec7326d9" />
